tests/bin/tap-mockexchangerates

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env sh
```

Comment:
This is a neat solution to running the test, but adding a dependency on `sh` seems like a bit much.

I'd like to keep this all in Python as much as possible, which, in this case can be accomplished by extracting the Popen call to a function that returns `tap`, and, in the test, replacing it with a mock function that just returns an object with the `stdout` property present that will provide the iterating interface expected by the rest of the code.

Does that make sense?